### PR TITLE
JBPM-7604 Stunner - Showing case management properties for cases only

### PR DESCRIPTION
**Pull request by tiagodolphine**
Hide / Show Case Management Properties whether the diagram is under a case project and the `adhoc` property is set to **true**

To test it is necessary 2 conditions:
- the existence of **.caseproject** file on the root project filesystem on the .niogit, on the WB just create a case project should be enough because it creates this file.
- the property adhoc set to true or false on the forms

Since the adhoc condition can be changed at runtime, you can test setting to true/false and see the Case Management properties be shown/hide.

@romartin 
@LuboTerifaj

### CODE REVIEW

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
@pefernan adding you as a reviewer as it updates some code on the Stunner side but related to forms... just please double check it makes sense for you. Thanks!

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/service/ProjectType.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+public enum ProjectType {
+
+    BPMN(null),
+    CASE(".caseproject");
+
+    ProjectType(String fileName) {
+        this.fileName = fileName;
+    }
+
+    private String fileName;
```
Comment (manstis) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/service/ProjectType.java]: Isn't it `fileExtension`?

Comment (tiagodolphine) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/service/ProjectType.java]: Hi @manstis in fact the file name is `.caseproject`, this is a trick part, you can see here https://github.com/kiegroup/jbpm-wb/blob/master/jbpm-wb-integration/jbpm-wb-integration-backend/src/main/java/org/jbpm/workbench/wi/backend/server/casemgmt/service/CaseProjectServiceImpl.java when a new Case Project is created on the WB, it creates this file, on the root of the repo, and this is used to identify when creating Assets under a case project. Anyway, this is the way it works, and this PR is to make Stunner bpmn to support case variables under a case project.

**Comment by tiagodolphine** [comment]
@romartin PR updated solving conflicts with master.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/service/ProjectType.java (hunk 1 above)
Comment (manstis) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/service/ProjectType.java]: Thanks for the explanation @tiagodolphine

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
jenkins retest this

**Comment by romartin** [comment]
Hey @tiagodolphine , I added just a single review comment, any feedback about it? I think it makes sense fixing it... Also still missing @LuboTerifaj 's review. Let me know whatever. thx!

**Comment by tiagodolphine** [comment]
Hey @romartin I saw your comment I didn't have time to take a deeper look, but I think if necessary I can improve that inserting a callback on the `loadDiagramType` combining with the `workItemDefinitionService.load`.  But in this way, it would be executed sequentially one after the other, the way it is today the calls to backend are being made in parallel and may be faster.

**Pull request merged**